core/templates/domain/objects/objects-domain.constants.spec.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview unit tests for the units object domain constants.
+ */
+
+import { ObjectsDomainConstants } from './objects-domain.constants';
+
+describe('ObjectsDomainConstants', () => {
+  const validUnits = [
```

Comment:
We can't have a source of truth like this in tests. It needs to be derived from existing constants in code, perhaps by combining the regular and currency lists. Otherwise you simply end up with more different lists that aren't connected and can go out of date.

Reply:
We do support all mathjs units + we define some currency units of our own, and I tried to find if there are any constants or functions mathjs exports so we can get a list of units, but I don't think they do.

Reply:
OK, then can you instead use something like unit.equals(unit) to check whether two units are equal rather than building an explicit mapping? https://mathjs.org/docs/datatypes/units.html#api

Reply:
I will try it. thanks!

Reply:
So I just tried it and it won't work as it converts them to their SI unit and then compares. as our use case is to allow all possible notations of a unit.

Reply:
Yup, that's a good idea. I was going to suggest raising an issue/question on their repo/discussions.

Reply:
Also, earlier comment still holds, we can't have something like this in tests. If this is blocked on the mathjs discussion then that might block this PR unless you can find an alternative way to solve it.

Reply:
So I asked about this and got the solution [here](https://github.com/josdejong/mathjs/discussions/3178), now we can remove the hard-coded allUnits array.

Reply:
We don't need to create an array of units, we can directly use ```math.Unit.isValuelessUnit(unit)``` to check.